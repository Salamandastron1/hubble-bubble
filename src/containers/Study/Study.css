.card-container {
  background-color: #990000;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
  padding: 2rem 0 2.5rem 0;
}