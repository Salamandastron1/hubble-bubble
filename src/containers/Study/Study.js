import React from 'react'
import { connect } from 'react-redux'
import Card from '../../components/Card/Card'
import './Study.css'

export const Study = props => {
  const { astronomicalObjects, isLoading } = props
  const cards = astronomicalObjects.map(star => {
    return <Card star={star} key={star.id}/>
  })
  if(isLoading) {
    return (
      <div className='spinner'>
        <div className='star1'></div>
        <div className='star2'></div>
      </div>
    )
  } else {
    return (
      <section className='card-container'>
        {cards}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  astronomicalObjects: state.astronomicalObjects,
  filter: state.filter,
  isLoading: state.isLoading
})

export default connect(mapStateToProps)(Study)