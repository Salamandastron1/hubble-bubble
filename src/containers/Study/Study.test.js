import { Study } from './Study'
import { shallow } from 'enzyme'
import React from 'react'
import { mapStateToProps } from './Study'

describe('study', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Study />)

    expect(wrapper).toMatchSnapshot()
  })
  describe('mapStateToProps', () => {
    it('should return an object with the properties desired from state', () => {
      const desired = {}
      const mockState = {}
      const result = mapStateToProps(mockState)

      expect(result).toEqual(desired)
    })
  })
})

