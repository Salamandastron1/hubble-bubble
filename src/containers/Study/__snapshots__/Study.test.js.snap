// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`study should match the snapshot 1`] = `
<section
  className="card-container"
/>
`;
